### DISCUSSION TITLE
PDF Files disappears on latex error

### DISCUSSION THREAD

**Discussion by henryliuser** [James-Yu/LaTeX-Workshop#3659] [Q&A] [answered]
I have rebuild on save and VSCode autosave enabled; i like this experience overall, but there's just a small issue I have sometimes. 

When I'm in the process of writing something and pause in between typing, the Latex syntax is incorrect and it will error on build. This deletes the pdf file that i'm currently viewing side-by-side (closes the pdf viewer and the entire side panel if it's the only tab), and then when i get it to build again i have to re-open the PDF and find the location where I left off.

Is there a way to just keep the old out-of-date pdf if the build fails, so that VSCode doesn't close the tab?

**Comment by James-Yu** [comment]
iirc failed compile does not remove the pdf. What’s your tools, recipes, and clean config?

**Reply by henryliuser** [reply]
Hi there, 

Can you show me how to give you that information? Sorry about that.

I can tell you that I'm on MacOS Monterey, using latest version of VSCode and just installed your extension. I haven't edited any of the default settings.

With some more investigation, it doesn't delete the pdf on most compilation failures. Only on some strange ones.
Here is a minimum file to reproduce

```latex
\documentclass[11pt]{article}

\begin{document}

Hello, world!

\newpage

% \begin{tabular}{r c l | l} {
%   a & a & a & a & a & a & a
% }

\end{document}
```

When you compile this, it obviously works fine. I open the PDF up to the side, and then when i uncomment the 3 lines and recompile, the PDF disappears. Strangely, this doesn't happen without the `\newpage` on my end.

**Comment by James-Yu** [comment, marked as answer]
The issue shall be resolved in the latest version `9.5.0`.

**Reply by henryliuser** [reply]
Just tried it, it's working now. And fixed some other cases that triggered the same bug too. 

Thank you! I appreciate your effort so much :)
